Make proxy endpoint configurable for s3 Http clients (#10332)
apache/iceberg: docs/docs/aws.md

For more details of configuration, see sections [URL Connection HTTP Client Configurations](#url-connection-http-client-configurations) and [Apache HTTP Client Configurations](#apache-http-client-configurations).

> Configurations for the HTTP client can be set in the catalog properties. The following are the available configurations:

Reviewer: ```suggestion
Configurations for the HTTP client can be set via catalog properties. Below is an overview of available configurations:
```
